File: android/src/main/java/com/jvanila/flutter/shopify/usecases/ForgotPasswordUseCase.java
package com.jvanila.flutter.shopify.usecases;

import com.jvanila.flutter.plugin.arch.PluginContext;
import com.shopapp.gateway.ApiCallback;
import com.shopapp.gateway.entity.Error;
import com.shopapp.shopify.api.ShopifyApi;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import kotlin.Unit;

public class ForgotPasswordUseCase extends ShopifyCallUseCase {

    private static final String ARG_EMAIL = "email";

    public ForgotPasswordUseCase(PluginContext<ShopifyApi> pluginContext) {
        super(pluginContext);
    }

    @Override
    protected void call(MethodCall input, final MethodChannel.Result result) {
        String email = input.argument(ARG_EMAIL);
        mPluginContext.api.instance.forgotPassword(email, new ApiCallback<Unit>() {
            @Override
            public void onResult(Unit unit) {
                result.success(true);
            }

            @Override
            public void onFailure(Error error) {
                System.out.println("onFailure -- " + error);
                result.error("ForgotPasswordUseCase", error.getMessage(), error);
            }
        });
    }
}

File: ios/Classes/UseCases/ForgotPasswordUseCase.swift
//
//  ForgotPasswordUseCase.swift
//  shopify
//
//  Created by Joseph jose on 10/09/19.
//

import UIKit
import ShopApp_Shopify
import ShopApp_Gateway
import Flutter

class ForgotPasswordUseCase: UseCase {

    static let ARG_EMAIL = "email"
    
    override public init(_ context: PluginContext) {
        super.init(context)
    }
    
    override public func trigger(with methodCall: FlutterMethodCall, result: @escaping (Any?) -> Void) {
        
        if let args = methodCall.arguments as? [String:String] {
            
            let email = args[ForgotPasswordUseCase.ARG_EMAIL]
            
            (mContext.api.instance as! ShopifyAPI).resetPassword(with: email!) { (success, error) in
                
                if error != nil {
                    
                    self.createError(withCase: "ForgotPasswordUseCase", message: error!.errorMessage, error: error!, on: result)
                }else {
                    
                    result(success)
                }
            }
        }
    }
}
